### PR TITLE
Fix LocationOfOrigin shows up unexpectedly

### LINKED ISSUE #117567: LocationOfOrigin shows up unexpectedly
### What happened?

yaml.Marshal a kubeconfig, as done in the example tests here https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api

### What did you expect to happen?

LocationOfOrigin is not seen in output, as comment says "never serialized."

### How can we reproduce it (as minimally and precisely as possible)?

Click "run example"

### Anything else we need to know?

`kubectl config view` doesn't suffer from this as it does some conversion

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind bug
#### Which issue(s) this PR fixes:

Fixes #117567

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/clientcmd/api/types.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ type Cluster struct {
 type AuthInfo struct {
 	// LocationOfOrigin indicates where this object came from.  It is used for round tripping config post-merge, but never serialized.
 	// +k8s:conversion-gen=false
-	LocationOfOrigin string
+	LocationOfOrigin string `json:"-"`
```
Comment (Tusenka) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/types.go:110]: Why we need to setup default value here?

**Comment by liggitt** [comment]
I thought the json tags on the internal type were needed because that's what the `kubectl config` json-path was evaluated against?

File: staging/src/k8s.io/client-go/tools/clientcmd/api/types.go [hunk 2]
```diff
@@ -69,19 +69,19 @@ type Cluster struct {
 	// +k8s:conversion-gen=false
 	LocationOfOrigin string
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/types.go:70]: Adding `json:"-"` would be a simpler way to resolve the linked issue

**Comment by wzshiming** [comment]
Updated.
I also added tags to the other 4 fields that had missed tags.

**Pull request merged**